Makefile.am

```diff
@@ -5,7 +5,9 @@
 ## Please see the COPYING and CONTRIBUTORS files for details.
 ##
 
-AUTOMAKE_OPTIONS = dist-bzip2 1.5 foreign
+# v1.13 has decent parallel-tests implementation. We require
+# parallel-tests (enabled by default in v1.13) because XXX.
+AUTOMAKE_OPTIONS = dist-bzip2 1.13 foreign
```

Comment:
This XXX placeholder will need to be filled in before merging this PR, of course: We must have a good reason for requiring Automake v1.13 and decent parallel-tests support does not seem sufficient (unless sequential tests -- default before v1.13 -- represent some non-obvious but significant problem for Squid). I have disclosed this new Automake version requirement in PR description.

Details (and alternative solutions) are discussed [earlier](https://github.com/squid-cache/squid/pull/1463#discussion_r1340475457).

If we have no reasons to require parallel-tests, we can undo the whole thing and remove the disclosure from PR description. This is one of the alternatives discussed earlier:

```suggestion
AUTOMAKE_OPTIONS = dist-bzip2 1.5 foreign
```

Reply:
Strictly speaking, we do not NEED parallel-tests. I suppose we can drop all requirements, and parallel tests will just happen if teh host system has a recent-enough automake.

This said, automake 1.13 was released in 2013, it predates the c++ standard we require by 4 years. Automake 1.12 even has a security vulnerability in it.

In other words, I'd keep the requirement, but we can also drop it.

Reply:
> we do not NEED parallel-tests. I suppose we can drop all requirements, and parallel tests will just happen if teh host system has a recent-enough automake.

The above matches my understanding as well.

> automake 1.13 was released in 2013, it predates the c++ standard we require by 4 years. Automake 1.12 even has a security vulnerability in it.

IMO, all that is pretty much irrelevant to our decision to require v1.13 -- it can only be used as icing on the cake: "We require v1.13 because XXX, especially since v1.13 is 10 years old and v1.12 has security vulnerabilities".

If Squid does not need v1.13, or we do not know whether it really needs v1.13, then I recommend that we (continue to) let admins decide what automake version they want to use. This PR is not really about any of that.

To avoid a misunderstanding, I am not trying to force you to make a particular decision here. I am OK with both "v1.13 for some documented/valid reason" or "leave as is".

Reply:
I'll leave as-is. It works, and admins can decide for themselves